app/controllers/secrets_controller.rb

```diff
@@ -54,6 +54,26 @@ def duplicates
       sort_by { |_, v| -v.size }
   end
 
+  def resolve
+    params.require([:project_id, :deploy_group, :keys])
+
+    project = Project.find(params[:project_id])
+    deploy_group = DeployGroup.find_by_permalink(params[:deploy_group])
+    keys = params[:keys]
+    resolver = Samson::Secrets::KeyResolver.new(project, [deploy_group])
+
+    resolved = {}
+    keys.each do |key|
+      resolved[key] = resolver.expand('unused', key).first&.last
```

Comment:
resolver returns multiple results when resolving a `_*` key, you might want to support that too and do an `each { |k, r| resolved[k] = r }`

Reply:
oh, i read it as it would be something like

```
{
  "wildcard_*": [
    {"wildcard_foo": "global/global/global/wildcard_foo" },
    {"wildcard_blah": "global/global/global/wildcard_blah" },
    ...
  ]
}
```

took your suggestion and merge them back into the hash, so it's kept flat

it looks like there's some implicit uppercasing going on though, but -- i left that as is. 🤷‍♂